manifests/plugin/discovery.pp

```diff
@@ -25,6 +25,8 @@
   if $install_images {
     $tftp_root_clean = regsubst($tftp_root, '/$', '')
 
+    ensure_resource('file', "${tftp_root_clean}/boot", {ensure => directory})
+
     foreman::remote_file {"${tftp_root_clean}/boot/${image_name}":
       remote_location => "${source_url}${image_name}",
       mode            => '0644',
```

Comment:
I think this should require that file resource then.

Reply:
Which is the oldest Puppet version supported by Foreman? Puppet 3? In Puppet 4 we have ordering = manifest as default option set.

Reply:
Hrm, that would be 4.6.2.

Reply:
Personally I prefer not to rely on manifest ordering

Reply:
OK. Done.